common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -239,9 +239,17 @@ private void captureGitMetadata() {
             String gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url");
             String gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD");
             String gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD");
-            String gitBranchName = execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD");
             String gitStatus = execAndGetStdOut("git", "status", "--porcelain");
 
+            String gitBranchName;
+            // It is okay to call System.getenv directly here since this lambda is saved to the configuration cache
+            // and re-executed on every build.
```

Comment:
interesting. for the standard case where we are calling the git command line, does that mean it will not be re-executed on every build?
i ask because this could mean that the git id changes and the old one will still be used as a custom value.

Reply:
The "lambda" is the one that starts on [line 234](https://github.com/gradle/gradle-enterprise-build-config-samples/pull/111/files#diff-e72bae4123cfd74f4e12e8d13bc2c9c5dff46d384cc8f3bbdd6224791c407553R234) and includes the calls to git.  

So the git commands are also called on every build (including the ones that get the git commit IDs).

Reply:
The latest commit converts the Lambda into a full static class which should make this more obvious.

I deleted the comment because I'm worried it might confuse users. I can add a similar comment back in if we think it would be useful.